ksql-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -94,17 +93,16 @@ private OutputNode buildOutputNode(final PlanNode sourcePlanNode) {
 
     final Into intoDataSource = analysis.getInto().get();
 
-    final Optional<Field> partitionByField = analysis.getPartitionBy()
-        .map(keyName -> inputSchema.findValueField(keyName)
+    final Optional<String> partitionByField = analysis.getPartitionBy();
+
+    partitionByField.ifPresent(keyName ->
```

Comment:
nit: I think this check may make more sense inside `buildOutputKeyField`

Reply:
Nar. This is checking the `PARTITION BY`  field is valid, not the key field.  If anything it would be in a `validatePartitionBy` method.

Reply:
derp 👍